src/Simplify.elm

```diff
@@ -4181,6 +4183,36 @@ listFoldAnyDirectionChecks foldOperationName checkInfo =
                                     []
 
 
+foldAndSetToListCompositionChecks : String -> CompositionCheckInfo -> List (Error {})
+foldAndSetToListCompositionChecks foldOperationName checkInfo =
+    let
+        ( earlier, later ) =
+            if checkInfo.fromLeftToRight then
+                ( checkInfo.left, checkInfo.right )
+
+            else
+                ( checkInfo.right, checkInfo.left )
+    in
+    case getSpecificFunctionCall ( [ "List" ], foldOperationName ) checkInfo.lookupTable later of
```

Comment:
I *think* we should be checking for the number of arguments here. Someone could write code like
```elm
List.foldr f << Set.toList -- instead of List.foldr f x << Set.toList
--> which would be transformed to
Set.foldr f
```
which wouldn't be correct I believe.